internal/querycoord/task.go

```diff
@@ -2485,11 +2472,7 @@ func (lbt *loadBalanceTask) globalPostExecute(ctx context.Context) error {
 		}
 	}
 
-	err := wg.Wait()
```

Comment:
Have to wait here to make sure the shard leader is updated before syncReplicaSegments()

Reply:
fixed!